app/code/core/Maho/Captcha/Model/Observer.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+/**
+ * Maho
+ *
+ * @package    Maho_Captcha
+ * @copyright  Copyright (c) 2025 Maho (https://mahocommerce.com)
+ * @license    https://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
+class Maho_Captcha_Model_Observer
+{
+    public function verify(Varien_Event_Observer $observer): void
+    {
+        $helper = Mage::helper('captcha');
+        if (!$helper->isEnabled()) {
+            return;
+        }
+
+        /** @var Mage_Core_Controller_Front_Action $controller */
+        $controller = $observer->getControllerAction();
+        $data = $controller->getRequest()->getPost();
+
+        $token = $data['maho_captcha'] ?? '';
+        if ($helper->verify((string) $token)) {
+            return;
+        }
+
+        $this->failedVerification($controller);
+    }
+
+    public function verifyAjax(Varien_Event_Observer $observer): void
```

Comment:
We should integrate the changes from #113 right?

Reply:
yep absolutely

Reply:
I'm trying to use the isAjax but it's not working :-\

Reply:
I pushed a commit that uses the original `<args><is_ajax>1</is_ajax></args>` idea from #113. I haven't had a chance to test it, but I think it's useful?

Reply:
yes it works but I've a couple of other problems, I'll write another comment in a bit